### PR TITLE
NamedArguments: fix false positive with java method call

### LINKED ISSUE #3289: NamedArguments with java code false positive
## Expected Behavior

This code should not raise any `NamedArguments` issue:
```kotlin
import org.joda.time.DateTime

DateTime(2020, 3, 13, 14, 0, 0)
```

## Observed Behavior
It does

## Context
If you use a java function in kotlin you can't use `NamedArguments` so there is no way to fix this.

## Your Environment
* Version of detekt used: 1.15.0-RC1

### PR DESCRIPTION
**Pull request by t-kameyama**
Fixes #3289

### CODE REVIEW

**Review by chao2zhang — APPROVED** [review]

**Comment by BraisGabin** [comment]
Are we OK about removing features of detekt without type solving? Or do we want to keep the mixed rules? I prefer no mix.

**Review by BraisGabin — APPROVED** [review]

**Review by cortinico — APPROVED** [review]

**Comment by cortinico** [comment]
> Are we OK about removing features of detekt without type solving? Or do we want to keep the mixed rules? I prefer no mix.

I'm fine with this change. I believe we should push users to use Type Resolution + restrict the rules with mixed behavior (#2994). This rule is going to become a Type Resolution only so it 👍  on my end.

File: detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NamedArguments.kt [hunk 1]
```diff
@@ -38,8 +41,12 @@ class NamedArguments(
     )
 
     override fun visitCallExpression(expression: KtCallExpression) {
+        if (bindingContext == BindingContext.EMPTY) return
         val valueArguments = expression.valueArguments
-        if (valueArguments.size > threshold && valueArguments.any { !it.isNamed() }) {
+        if (valueArguments.size > threshold &&
+            valueArguments.any { !it.isNamed() } &&
+            expression.getResolvedCall(bindingContext)?.candidateDescriptor?.hasStableParameterNames() == true
```
Comment (schalkms) [inline detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NamedArguments.kt:48]: That's a nice solution. Thank you for that.
Do you know by chance, whether there exists an API to detect Java method calls?

File: detekt-rules-complexity/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NamedArgumentsSpec.kt [hunk 2]
```diff
@@ -107,7 +111,19 @@ class NamedArgumentsSpec : Spek({
                 
                 val obj = C(1, 2)
             """
-            val findings = namedArguments.compileAndLint(code)
+            val findings = namedArguments.compileAndLintWithContext(env, code)
+            assertThat(findings).hasSize(0)
+        }
+
+        it("java method invocation") {
```
Comment (schalkms) [inline detekt-rules-complexity/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NamedArgumentsSpec.kt:118]:
```suggestion
        it("java method invocation should not be flagged") {
```

**Review by schalkms — APPROVED** [review]

**Comment by schalkms** [comment]
I agree. This shouldn't be a mixed rule.
The goal is to stay away from that, since type and symbol solving should become stable with detekt v2.0.0.

File: detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NamedArguments.kt (hunk 1 above)
Comment (t-kameyama) [reply detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NamedArguments.kt:48]: `expression.getResolvedCall(bindingContext)?.candidateDescriptor is JavaMethodDescriptor`

Comment (schalkms) [reply detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NamedArguments.kt:48]: cool, thanks

**Pull request merged**